@xen-orchestra/backups/_deltaVm.js

```diff
@@ -165,6 +166,12 @@ exports.importDeltaVm = defer(async function importDeltaVm(
     }
   }
 
+  if (!isEmpty(mapVdisSrs)) {
+    for (const [vdiUuid, srId] of Object.entries(mapVdisSrs)) {
+      mapVdisSrs[vdiUuid] = xapi.getObject(srId, 'SR').$ref
```

Comment:
Let's only deal with API objects here, not XO ones.

XAPI objects have no `id`, let's make it clear that this code only supports `UUID`s.

```suggestion
    for (const [vdiUuid, srUuid] of Object.entries(mapVdisSrs)) {
      mapVdisSrs[vdiUuid] = xapi.getObject(srUuid, 'SR').$ref
```

Reply:
`Xapi#getObject` does not accept a type as second param.

Reply:
Let's make this code as little dependant as possible to cached records, you can do something like:

```js
const CACHE = new Map()
async function resolveUuid(uuid, type) {
  let ref = CACHE.get(uuid)
  if (ref === undefined) {
    ref = await xapi.call(`${type}.get_by_uuid`)
    CACHE.set(uuid, ref)
  }
  return ref
}
```
```suggestion
    for (const [vdiUuid, srUuid] of Object.entries(mapVdisSrs)) {
      mapVdisSrs[vdiUuid] = await resolveUuid(srUuid, 'SR')
    }
```